pandas/core/nanops.py

```diff
@@ -60,8 +60,11 @@ def __call__(self, f):
         def _f(*args, **kwargs):
             obj_iter = itertools.chain(args, kwargs.values())
             if any(self.check(obj) for obj in obj_iter):
-                msg = "reduction operation {name!r} not allowed for this dtype"
-                raise TypeError(msg.format(name=f.__name__.replace("nan", "")))
+                msg = (
+                    f"reduction operation {f.__name__.replace('nan', '')!r}"
```

Comment:
use repr rather than ``!r``

Reply:
Sure, I saw another issue regarding !r, thought I shouldn't replace it with this PR.

Reply:
neh its fine to do here